haiopy/buffers.py

```diff
@@ -216,12 +217,13 @@ def data(self):
     def data(self, data):
         """Set the underlying signal if the buffer is not active."""
         self.check_if_active()
+        self._n_samples = data.n_samples
         self._data = self._pad_data(data)
         self._update_data()
 
     def _set_block_size(self, block_size):
         super()._set_block_size(block_size)
-        self._update_data()
+        self.data = self._data[:self._n_samples]
```

Comment:
```suggestion
        self.data.time = self._data.time[..., :self._n_samples]
```
I think this may not work as intended. Slicing a Signal will slice along the channel axes, not the time samples of the underlying time data.